api/tests/integration/AuthTest.php

```diff
@@ -274,4 +278,167 @@ public function testMakeLoginToken()
 
         $this->assertEquals($id, $user->user_id);
     }
+
+    public function testEnvironmentForSocial()
+    {
+        // Make sure we have the hosts env variable for the redirect url generation
+        $hostApi = $this->app['config']['hosts.api'];
+        $hostApp = $this->app['config']['hosts.app'];
+
+        $this->assertNotNull($hostApi);
+        $this->assertStringStartsWith('http', $hostApi);
+        $this->assertStringEndsNotWith('/', $hostApi);
+        $this->assertNotNull($hostApp);
+        $this->assertStringStartsWith('http', $hostApp);
+        $this->assertStringEndsNotWith('/', $hostApp);
+    }
+
+    public function testInvalidProvider()
+    {
+        $this->get('/auth/social/foobar');
+
+        $this->assertException('provider', 404, 'NotFoundHttpException');
+    }
+
+    public function testProviderRedirect()
+    {
+        $this->get('/auth/social/facebook');
+
+        $this->assertResponseStatus(302);
+        $this->assertContains('refresh', $this->response->getContent());
```

Comment:
Ah understood, yea that isn't needed as our server can interpret the location header. Well the frontend can only support >= IE9 so I think we're safe to not require that.